File: Assets/Scripts/changeSprite.js
#pragma strict
var sprite1 : Sprite;
var sprite2 : Sprite;


function OnCollisionEnter2D(coll : Collision2D) {
	if (coll.gameObject.tag == "Player")
    {
			var currentSprite = GetComponent(SpriteRenderer).sprite;
			if (currentSprite == sprite1) {
				GetComponent(SpriteRenderer).sprite = sprite2;
			}	 else {
				GetComponent(SpriteRenderer).sprite = sprite1;
			}
    }
}

File: Assets/Scripts/changeSprite.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class changeSprite : MonoBehaviour
{
    public Sprite sprite1;
    public Sprite sprite2;
    public virtual void OnCollisionEnter2D(Collision2D coll)
    {
        if (coll.gameObject.tag == "Player")
        {
            Sprite currentSprite = ((SpriteRenderer) this.GetComponent(typeof(SpriteRenderer))).sprite;
            if (currentSprite == this.sprite1)
            {
                ((SpriteRenderer) this.GetComponent(typeof(SpriteRenderer))).sprite = this.sprite2;
            }
            else
            {
                ((SpriteRenderer) this.GetComponent(typeof(SpriteRenderer))).sprite = this.sprite1;
            }
        }
    }

}